### PR TITLE
Fish

### PR DESCRIPTION
**Pull request by fadwil**
__x__ Wrote Tests ____ Implemented ____ Reviewed

Neccesary checkmarks:

    [x] All Tests are Passing

    [] The code will run locally

Type of change

    [x] New feature
    [] Bug Fix

Implements/Fixes:

    description closes #

Check the correct boxes

    [x] This broke nothing
    [] This broke some stuff
    [] This broke everything

Testing Changes

    [] No Tests have been changed
    [x] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)

Checklist:

    [x] My code has no unused/commented out code
    [x] I have reviewed my code
    [] I have commented my code, particularly in hard-to-understand areas
    [x] I have fully tested my code

(For Fun!)Please Include a link to a gif of your feelings about this branch
https://media.giphy.com/media/G5tA3ZSgiBiIo/giphy.gif
Link:

### CODE REVIEW

File: app/controllers/api/v1/fish_controller.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+class Api::V1::FishController < ApplicationController
+  before_action :set_user
+  before_action :set_fish, only: [:update, :show]
+
```
Comment (Daniel-Gallagher92) [inline app/controllers/api/v1/fish_controller.rb:4]: Nice use of the before actions to dry out the controller actions.

File: app/controllers/api/v1/fish_controller.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  rescue_from ActiveRecord::RecordNotFound, with: :record_not_found
```
Comment (Daniel-Gallagher92) [inline app/controllers/api/v1/fish_controller.rb:5]: does this prevent us from needing begin/rescue in the show/index actions?

File: app/controllers/api/v1/fish_controller.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  def fish_params
+    params.require(:fish).permit(:species, :weight, :length)
+  end
+
+  def record_not_found(error)
+    # Extracts model name from the error message
+    model_name = error.message.split(" ")[2..2].join(" ").gsub(/#/, '').singularize
+  
+    render json: { error: "No #{model_name} with that ID could be found" }, status: :not_found
+  end
```
Comment (Daniel-Gallagher92) [inline app/controllers/api/v1/fish_controller.rb:59]: Loving how flexible this method makes the error handling

File: app/controllers/api/v1/lures_controller.rb [hunk 4]
```diff
@@ -54,7 +54,6 @@ def lure_params
   end
 
   def record_not_found(error)
-    # require 'pry'; binding.pry
```
Comment (Daniel-Gallagher92) [inline app/controllers/api/v1/lures_controller.rb:57]: lolololol, thanks for finding that.

File: spec/requests/api/v1/users/fish/delete_fish_spec.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  describe "DELETE /api/v1/users/:user_id/fish/:id" do
+    it "can delete a fish w/ valid params" do
+      user = create(:user)
+      fish = user.fish.create(species: "Walleye", weight: 1.1, length: 12.1)
+
+      headers = { "CONTENT_TYPE" => "application/json" }
+      delete "/api/v1/users/#{user.id}/fish/#{fish.id}", headers: headers
+
+      expect(response).to be_successful
+      expect(response.status).to eq(200)
```
Comment (Daniel-Gallagher92) [inline spec/requests/api/v1/users/fish/delete_fish_spec.rb:13]: Should a successful delete return a 204 no content?

**Review by Daniel-Gallagher92 — APPROVED** [review]
Disregard my comment on the 204 error, you're absolutely right that a successful delete should return. 200 'ok'. Everything else looks solid, thank you for cleaning up some messy areas.

**Review by Daniel-Gallagher92 — APPROVED** [review]

File: spec/requests/api/v1/users/fish/delete_fish_spec.rb (hunk 5 above)
Comment (fadwil) [reply spec/requests/api/v1/users/fish/delete_fish_spec.rb:13]: I was thinking that, I noticed its a 200 in the lures delete as well, probably need to refactor both of these

Comment (fadwil) [reply spec/requests/api/v1/users/fish/delete_fish_spec.rb:13]: oh I see your below comment about the 204, got it

**Pull request merged**